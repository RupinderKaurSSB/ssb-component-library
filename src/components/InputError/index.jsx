import React from 'react';
import PropTypes from 'prop-types';

const InputError = ({ className, errorMessage, negative }) => (
	<div className={`ssb-input-error${negative ? ' negative' : ''}${className ? ` ${className}` : ''}`}>
		<span>{errorMessage}</span>
	</div>
);

InputError.defaultProps = {
	negative: false,
};

InputError.propTypes = {
	className: PropTypes.string,
	errorMessage: PropTypes.string.isRequired,
	negative: PropTypes.bool,
};

export default InputError;
