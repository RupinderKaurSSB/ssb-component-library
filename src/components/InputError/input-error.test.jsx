import React from 'react';
import {shallow} from 'enzyme';
import InputError from './index';

describe('InputError component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<InputError errorMessage="Insert error message" />);
		expect(wrapper).toMatchSnapshot();
	});
	test('Toggles classNames correctly', () => {
		const wrapper = shallow(<InputError errorMessage="Insert error message" negative />);
		expect(wrapper.hasClass('negative')).toEqual(true);
	});
});
