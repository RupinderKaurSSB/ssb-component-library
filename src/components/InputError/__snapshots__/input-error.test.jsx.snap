// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputError component Matches the snapshot 1`] = `
<span
  className="ssb-input-error "
>
  Insert error message
</span>
`;
